extractor/src/main/java/org/schabi/newpipe/extractor/utils/ProtoBuilder.java

```diff
@@ -0,0 +1,72 @@
+package org.schabi.newpipe.extractor.utils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+public class ProtoBuilder {
+    ByteArrayOutputStream byteBuffer;
+
+    public ProtoBuilder() {
+        this.byteBuffer = new ByteArrayOutputStream();
+    }
+
+    public byte[] toBytes() {
+        return byteBuffer.toByteArray();
+    }
+
+    public String toUrlencodedBase64() {
+        final String b64 = Base64.getUrlEncoder().encodeToString(toBytes());
+        return URLEncoder.encode(b64, StandardCharsets.UTF_8);
+    }
+
+    private void writeVarint(final long val) {
+        try {
+            if (val == 0) {
+                byteBuffer.write(new byte[]{(byte) 0});
+            } else {
+                long v = val;
+                while (v != 0) {
+                    byte b = (byte) (v & 0x7f);
+                    v >>= 7;
+
+                    if (v != 0) {
+                        b |= (byte) 0x80;
+                    }
+                    byteBuffer.write(new byte[]{b});
+                }
+            }
+        } catch (final IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Side note: It may be better to use ``UncheckedIOException`` here since it's more specific
This might be also relevant for the rest of the file ;)